scib/metrics/lisi.py

```diff
@@ -188,6 +180,13 @@ def lisi_graph_py(
     Compute LISI score on shortes path based on kNN graph provided in the adata object.
     By default, perplexity is chosen as 1/3 * number of nearest neighbours in the knn-graph.
     """
+    try:
+        import rpy2.rinterface_lib.callbacks
+        import rpy2.rinterface_lib.embedded
+
+        rpy2.rinterface_lib.callbacks.logger.setLevel(logging.ERROR)
+    except ModuleNotFoundError as e:
+        raise OptionalDependencyNotInstalled(e)
```

Comment:
I don't think this is needed here. the only rpy2 command is an RRuntimeError, which is actually not needed anymore as it's all computed in python and C++, no? Need to check with @mbuttner, but this looks like legacy rpy2 code to me and so rpy2 isn't needed for lisi computation at all.

Reply:
True, we wouldn't expect an R Runtime error. Do we still want to keep the behaviour of LISI returning NaN when the kNN graph fails?

Reply:
LISI does not need any rpy2 anymore. As @LuckyMD said, it is all Python and C++. The old LISI implementation needed R.

Reply:
thanks for confirmation, @mbuttner ! And @mumichae: yes, NaN when kNN fails is good.

Reply:
OK, I'm catching a `RuntimeError` now instead

Reply:
Cool!